pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,1495 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+//! Nodle Chain - custom tailored, staking pallet.
+//! Use a non inflationary reward system.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(any(feature = "runtime-benchmarks", test))]
+mod benchmarking;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+mod set;
+pub mod weights;
+
+use frame_support::pallet;
+pub(crate) mod hooks;
+pub mod migrations;
+pub(crate) mod slashing;
+pub(crate) mod types;
+
+#[cfg(feature = "std")]
+use frame_support::traits::GenesisBuild;
+
+pub use pallet::*;
+
+#[pallet]
+pub mod pallet {
+    use super::*;
+    use crate::set::OrderedSet;
+    use frame_support::{
+        pallet_prelude::*,
+        traits::{
+            Currency, ExistenceRequirement, Get, Imbalance, LockIdentifier, LockableCurrency,
+            OnUnbalanced, WithdrawReasons,
+        },
+    };
+    use frame_system::pallet_prelude::*;
+    use frame_system::{self as system};
+    use sp_runtime::{
+        traits::{AccountIdConversion, Saturating, Zero},
+        DispatchResult, ModuleId, Perbill,
+    };
+    use sp_staking::SessionIndex;
+    use sp_std::{convert::From, prelude::*};
+
+    pub use weights::WeightInfo;
+
+    use types::{Bond, Nominator, RewardPoint, SpanIndex, UnappliedSlash, UnlockChunk, Validator};
+
+    pub use types::{ValidatorSnapshot, ValidatorSnapshotOf};
+
+    pub use hooks::{SessionInterface, StashOf};
+
+    pub(crate) type BalanceOf<T> =
+        <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+    pub(crate) type PositiveImbalanceOf<T> = <<T as Config>::Currency as Currency<
+        <T as frame_system::Config>::AccountId,
+    >>::PositiveImbalance;
+
+    pub(crate) type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
+        <T as frame_system::Config>::AccountId,
+    >>::NegativeImbalance;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        /// The staking balance.
+        type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+        /// Handler for the unbalanced reduction when slashing a staker.
+        type Slash: OnUnbalanced<NegativeImbalanceOf<Self>>;
+        /// Handler for the unbalanced increment when rewarding a staker.
+        type Reward: OnUnbalanced<PositiveImbalanceOf<Self>>;
+        /// Number of sessions that staked fund remain bonded for
+        type BondedDuration: Get<SessionIndex>;
```

Comment:
```suggestion
        /// The staking balance.
        type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
        /// Number of sessions that staked fund remain bonded for
        type BondedDuration: Get<SessionIndex>;
```
I don't think those two are used anymore

Reply:
Kept Slash, but removed reward, Done :+1: